Sources/StreamChat/Query/QueryOptions.swift

```diff
@@ -5,32 +5,32 @@
 import Foundation
 
 /// Query options.
-struct QueryOptions: OptionSet, Encodable {
+public struct QueryOptions: OptionSet, Encodable {
```

Comment:
@b-onc why are we making it public? what would be the use case?

Reply:
There are more than 1 case, integrators currently can't pass their own query options, which is something allowed by the API.
1. Integrators can't query with `presence` option, which is something required for getting presence events.
2. Integrators can't query without `watch` option, for cases when they only want the data and not WS events.

Reply:
@bahadir ok, didn’t know it’s allowed to pass custom query options 🙂 the thing is with the current shape of offline support allowing to query channels without the `watch` option/allowing to stop watching a channel is dangerous. The local state will go out of sync much more often which will cause another spike of conflicting updates on both channel/message list

Reply:
@evsaev that's exactly why we didn't made them `public` firstly but now customers require it

Reply:
@b-onc Could we align this to the same approach that we applied with `layout options`? We basically get rid of the `OptionSet`. Since we're turning the `struct` public, we should make sure the way how you can extend the options is consistent.

Reply:
You can't send custom options, options are well-defined in the API: https://getstream.io/chat/docs/rest/#channels-getorcreatechannel_type_id
`watch`,`state` and `presence` is allowed

Reply:
In this case `QueryOptions.init` should stay internal, right?

Reply:
correct, good catch!

Reply:
@evsaev oh, it's public because of the `RawRepresentable` conformance... Alright, we should get rid of `OptionSet` as we did with `LayoutOptions`

Reply:
@evsaev @bielikb I've worked on this but i think resulting API is not better than what we have now. Apple also exposes inits for their OptionSet s but that doesn't mean developers should use them. even if they did, it won't work.
in our case, only allowed values will be encoded to queries, the other will be discarded. so if one used the QueryOptions.init(rawValue: 10000) it would just be a no-op, as it should be.
hiding that init requires a lot more "workarounds" while having the OptionSet API, I think we should not touch it. it's fundamentally different than LayoutOptions in which, no customization is allowed, QueryOptions is just communication API between backend, not between components inside the SDK

Reply:
Fair enough @b-onc !